### PR TITLE
refactor: Migrate touch actions to Kotlin

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/KeyDispatch.kt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    return dispatchKeyEvent(dispatcherAdapter, actionObject, inputState, inputStateTable, tickDuration, false)
+}
+
+/**
+ * Helper method to get a Key Event that has common attributes between KeyUp and KeyDispatch
+ * @param rawKey The value of the key stroke
+ * @return W3CKeyEvent Key event info used by adapter
+ */
+@Throws(AppiumException::class)
+fun getKeyEvent(dispatcherAdapter: W3CActionAdapter, rawKey: String?): W3CKeyEvent {
```
Comment (KazuCocoa) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/KeyDispatch.kt:134]: Can write like this?
```kotlin
fun getKeyEvent(dispatcherAdapter: W3CActionAdapter, rawKey: String?): W3CKeyEvent {
    return W3CKeyEvent().apply {
        // 1. Let raw key be action's value property
        // 2. Let key be equal to the normalised key value for raw key
        this.key = instance!!.getNormalizedKey(rawKey!!)

        // 4. Let code be 'code' for raw keypress
        this.code =  getCode(rawKey)

        // 5. Let location be the key location for raw key
        this.location = getLocation(rawKey)

        // 6. Let charCode, keyCode and which be the implementation-specific values of the charCode,
        // keyCode and which properties appropriate for a key with key key and location location on
        // a 102 key US keyboard
        this.keyCode = dispatcherAdapter.getKeyCode(this.key, this.location)
        this.charCode = dispatcherAdapter.getCharCode(this.code, this.location)
        this.which = dispatcherAdapter.getWhich(this.key, this.location)
    }
```

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/MotionEventBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+import androidx.test.espresso.InjectEventSecurityException
+import androidx.test.espresso.UiController
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import io.appium.espressoserver.lib.handlers.exceptions.MoveTargetOutOfBoundsException
+import io.appium.espressoserver.lib.helpers.AndroidLogger.Companion.logger
+import io.appium.espressoserver.lib.helpers.w3c.models.InputSource
+
+class MotionEventBuilder {
+    private val motionEventParams: MotionEventParams
+    fun withX(x: List<Long>?): MotionEventBuilder {
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/MotionEventBuilder.kt]: changed

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/MotionEventBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+            // Don't do anything if no pointers were provided
+            if (pointerCount == 0 && motionEventParams.action != MotionEvent.ACTION_CANCEL) {
+                return null
+            }
+            val pointerCoords = arrayOfNulls<MotionEvent.PointerCoords>(pointerCount)
+            val pointerProperties = arrayOfNulls<PointerProperties>(pointerCount)
+            for (pointerIndex in 0 until pointerCount) {
+                // Set pointer coordinates
+                pointerCoords[pointerIndex] = MotionEvent.PointerCoords()
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/MotionEventBuilder.kt:64]: changed

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/KeyDispatch.kt (hunk 1 above)
Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/KeyDispatch.kt]: done, I did not know about `apply`

Comment (KazuCocoa) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/KeyDispatch.kt]: 👍 
It is helpful to get the class itself (as `return this`) :)

**Pull request merged**